sdk/search/Azure.Search.Documents/tests/Samples/Sample01_HelloWorld.cs

```diff
@@ -466,5 +466,62 @@ public async Task GetCountAsync()
             Console.WriteLine($"Search index {indexName} has {count.Value} documents.");
             #endregion Snippet:Azure_Search_Tests_Samples_GetCountAsync
         }
+
+        [Test]
+        public async Task QuerySession()
+        {
+            const int size = 150;
+            await using SearchResources resources = await SearchResources.CreateLargeHotelsIndexAsync(this, size, false, true);
+            Environment.SetEnvironmentVariable("SEARCH_ENDPOINT", resources.Endpoint.ToString());
+            Environment.SetEnvironmentVariable("SEARCH_API_KEY", resources.PrimaryApiKey);
+            Environment.SetEnvironmentVariable("SEARCH_INDEX", resources.IndexName);
+
+            HashSet<string> uniqueDocuments = new HashSet<string>();
+            bool hasDuplicates = false;
+
+            #region Snippet:Azure_Search_Tests_Samples_QuerySession
+            Uri endpoint = new Uri(Environment.GetEnvironmentVariable("SEARCH_ENDPOINT"));
+            AzureKeyCredential credential = new AzureKeyCredential(Environment.GetEnvironmentVariable("SEARCH_API_KEY"));
+            string indexName = Environment.GetEnvironmentVariable("SEARCH_INDEX");
+
+            SearchClient searchClient = new SearchClient(endpoint, indexName, credential);
+#if !SNIPPET
+            searchClient = InstrumentClient(new SearchClient(endpoint, indexName, credential, GetSearchClientOptions()));
+#endif
+            SearchResults<Hotel> results = await searchClient.SearchAsync<Hotel>(
+                    new SearchOptions
+                    {
+                        Filter = "Rating gt 2",
+                        SessionId = "Session-1"
+                    });
+
+            int totalDocsCount = 0;
+            int pageCount = 0;
+            await foreach (Page<SearchResult<Hotel>> page in results.GetResultsAsync().AsPages())
```

Comment:
Why are we enumerating pages? This is not typically needed and complicates the example. Just `await foreach` over the whole collection unless you really need to query pages. Is it only to get page count? I wouldn't think most people would care. Our samples should show the mainline cases unless specifically for non-mainline cases, which as a "Hello world" sample doesn't seem appropriate here.